src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfoOfT.WriteHelpers.cs

```diff
@@ -111,11 +117,23 @@ public partial class JsonTypeInfo<T>
                         Utf8JsonWriterCache.ReturnWriter(writer);
                     }
 
-                    await serializationContext.FlushAsync(cancellationToken).ConfigureAwait(false);
+                    FlushResult result = await pipeWriter.FlushAsync(cancellationToken).ConfigureAwait(false);
+                    if (result.IsCanceled || result.IsCompleted)
+                    {
+                        if (result.IsCanceled)
+                        {
+                            ThrowHelper.ThrowOperationCanceledException_PipeWriteCanceled();
+                        }
+
+                        ThrowHelper.ThrowOperationCanceledException_PipeWriteCompleted();
```

Comment:
Let's stop throwing here. I explained my reasoning in https://github.com/dotnet/runtime/pull/101461#discussion_r1610622231.

If we want to special case completed FlushResults, we could stop serialization early and return a completed Task. This should have no observable affect other than saving cycles since the completed flush result implies that the reader has already stopped reading.

I think ignoring the completed FlushResult is fine for now though.